store/src/state.rs

```diff
@@ -266,7 +266,39 @@ impl State {
 
             drop(guard);
 
-            (account_tree, chain_mmr, nullifier_tree)
+            let notes = notes
+                .iter()
+                .enumerate()
+                .map(|(count, note)| {
+                    // The Merkle tree leaves are interleaved note data with note metadata. The
+                    // note's are on the even indexes.
+                    let note_index = count * 2;
```

Comment:
we should use `note.note_index` here right? And so no need for the `enumerate()` anymore

Reply:
oh, true, i forgot about this one. thanks!